app/code/Magento/Payment/Model/Method/Adapter.php

```diff
@@ -508,11 +508,12 @@ private function executeCommand($commandCode, array $arguments = [])
         }
 
         if (isset($arguments['payment'])) {
+            $payment = $arguments['payment'];
             $arguments['payment'] = $this->paymentDataObjectFactory->create($arguments['payment']);
-        }
-
-        if ($this->commandExecutor !== null) {
-            return $this->commandExecutor->executeByCode($commandCode, $arguments);
+            
+            if ($this->commandExecutor !== null) {
```

Comment:
According to signature `public function executeByCode($commandCode, InfoInterface $payment = null, array $arguments = [])` it's possible to pass `null` as second argument. If `InfoInterface` instance is required later in code then there should be thrown exception. But according to interfaces it's valid case when `$arguments` doesn't contains `payment`.